### PR TITLE
Test cmake

### PR DESCRIPTION
**Pull request by NathanW2**
## Description
Refactors the cmake test logic so you only have to update the TESTS variable with the file name and the rest is take care of. 

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: tests/src/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,34 @@
...
+    SET(PLATFORM_LIBRARIES wsock32)
+  ENDIF (WIN32)
+
+  # Since the tests are not actually installed, but rather
+  # run directly from the build/src/tests dir we need to
+  # ensure the qgis libs can be found.
+  IF (APPLE)
+  # For Mac OS X, the executable must be at the root of the bundle's executable folder
+  #  SET (CMAKE_INSTALL_NAME_DIR @executable_path/../../../src/core)
+  ENDIF (APPLE)
```
Comment (m-kuhn) [inline tests/src/CMakeLists.txt]: Is this block still required?

File: tests/src/analysis/CMakeLists.txt [hunk 2]
```diff
@@ -61,27 +36,38 @@ ENDIF (APPLE)
...
+  SET(${TESTNAME}_SRCS ${TESTSRC} ${util_SRCS})
+  SET(${TESTNAME}_MOC_CPPS ${TESTSRC})
+  ADD_EXECUTABLE(${TESTNAME} ${${TESTNAME}_SRCS})
+  SET_TARGET_PROPERTIES(${TESTNAME} PROPERTIES AUTOMOC TRUE)
+  TARGET_LINK_LIBRARIES(${TESTNAME}
     ${QT_QTCORE_LIBRARY}
     ${QT_QTTEST_LIBRARY}
     qgis_analysis)
-  ADD_TEST(qgis_${testname} ${CMAKE_CURRENT_BINARY_DIR}/../../../output/bin/qgis_${testname} -maxwarnings 10000)
+  ADD_TEST(${TESTNAME} ${CMAKE_CURRENT_BINARY_DIR}/../../../output/bin/${TESTNAME} -maxwarnings 10000)
```
Comment (m-kuhn) [inline tests/src/analysis/CMakeLists.txt]: Could that be

    ADD_TEST(${TESTNAME} ${CMAKE_BINARY_DIR}/output/bin/${TESTNAME} -maxwarnings 10000)

I think sometimes abosolute is better than relative ;)

**Comment by m-kuhn** [comment]
I like it 👍 
Comments are not concerning your changes, only general observations which could be changed if you agree.

**Review by m-kuhn — APPROVED** [review]

File: tests/src/CMakeLists.txt (hunk 1 above)
Comment (NathanW2) [inline tests/src/CMakeLists.txt]: I suspect not but was just going to leave as a place holder for future if needed.

**Pull request merged**